### DISCUSSION TITLE
Hire Plus Server in WPF application

### DISCUSSION THREAD

**Discussion by nedaalizadeh** [PlusToolkit/PlusLib#814] [Q&A] [answered]
Hello,
I'm currently working on WPF application that connect to Flir cameras and use the Spinnaker SDK to communicate with them. I want to show live video from cameras in my WPF application by using the Plus server and Directshow (or other proper video pllayer). I've read the Plus Toolkit document but I couldn't find out how to do that. Anyone can help me? Is it possible?

**Comment by adamrankin** [comment]
My recommendation would be to learn from an existing application.

This is the entry point for our most commonly used program, and provides a good overview of the classes necessary to collect data:
https://github.com/PlusToolkit/PlusLib/blob/master/src/PlusServer/Tools/PlusServer.cxx

This line:
https://github.com/PlusToolkit/PlusLib/blob/master/src/PlusServer/Tools/PlusServer.cxx#L164

is where you would put code to query the data collector for the latest frames, and then use them in your program as you see fit (probably extracting the video memory, and copying it to your WPF UI component).

**Comment by lassoan** [comment]
Another option would be that use PlusServer as is and just implement receiving data from a socket using OpenIGTLink protocol. You can either use the OpenIGTLink library in your application or (since OpenIGTLink is a very simple protocol) you can implement message receiving and unpacking from scratch.

**Comment by nedaalizadeh** [comment, marked as answer]
Thanks for your answers. I am going to implement and test both solutions. :)
